compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -413,6 +413,7 @@ pub trait Machine<'tcx>: Sized {
         _tcx: TyCtxtAt<'tcx>,
         _machine: &Self,
         _alloc_extra: &Self::AllocExtra,
+        _ptr: Pointer<Option<Self::Provenance>>,
```

Comment:
Document that these pointers are always "the same" information as the `prov` argument, just before being taken apart